README.md

```diff
@@ -30,7 +30,7 @@ For Linux and macOS binaries see the [GitHub release assets](https://github.com/
 
 ## Developer Installation
 
-If you have Go 1.16+, you can setup a development environment:
+If you have Go 1.17+, you can setup a development environment:
```

Comment:
if we bump here in the readme, we should change the go module as well

cc @mattmoor @dlorenc

Reply:
I'm not an expert on Go modules but AFAICT the [Go version in `go.mod`](https://go.dev/ref/mod#go-mod-file-go) has 2 effects:

1. Restricts language features introduced after that version.
2. Changes the behavior of the `go` command. In particular, for the 1.16->1.17 transition the consequential changes:
    - starts including explicit `requires` in `go.mod` for transitive dependencies
    - vendoring acts a little different (starts recording dependencies' Go versions from `go.mod`, omits `go.mod` and `go.sum` for vendored dependencies)

Because as of right now we still want to support 1.16, my initial thought is that (1) is a good thing for development and I'm inclined not to change it. Of course, will defer to you all :)

Reply:
We chatted about bumping the other one to 1.17 soon, this one seems fine as is and we can do the other one later.